migration/sql-files/062-workitem-related-changes.sql

```diff
@@ -0,0 +1,77 @@
+-- add a column to record the timestamp of the latest addition/change/removal of a comment on a workitem
+ALTER TABLE work_items ADD COLUMN commented_at timestamp with time zone;
+
+-- trigger to fill the `commented_at` column when a comment is added, updated or removed (soft delete, it's a record update)
+CREATE FUNCTION workitem_comment_insert_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.created_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION workitem_comment_update_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.updated_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION workitem_comment_softdelete_timestamp() RETURNS trigger AS $$
+    BEGIN
+        UPDATE work_items wi SET commented_at = NEW.deleted_at WHERE wi.id::text = NEW.parent_id;
+        RETURN NEW;
+    END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER workitem_comment_insert_trigger AFTER INSERT ON comments 
```

Comment:
We have to be careful with triggers. If for testing purposes you insert a soft deleted comment that shall not appear in the UI, that will update the `commented_at` field as well. I would like to see that the triggers take care of not updating when inserting a deleted comment (`deleted_at IS NOT NIL`).